### PR TITLE
MAX-11773 Add config for proxy setup and split chunks

### PR DESCRIPTION
**Pull request by elvinfucom**
Our latest react-scripts can't work well on app. Add config for proxy setup and split chunks to fix the issues.
**Proxy setup:**
Added proxy setup to `webpackDevServer.config.js`

See more
- Proxying API Requests in Development
https://facebook.github.io/create-react-app/docs/proxying-api-requests-in-development
- see section:  Move advanced proxy configuration to src/setupProxy.js
https://github.com/facebook/create-react-app/issues/5103#issue-364148526

**Split chunks**
Added `ENABLE_SPLIT_CHUNKS` and `GENERATE_RUNTIME_CHUNK` to allow disable split/runtime chunks based on app needs.

### CODE REVIEW

**Comment by elvinfucom** [comment]
@joshsylvester @mallik-svmx @sheela-svmx please help review.

File: packages/react-scripts/config/webpack.config.prod.js [hunk 1]
```diff
@@ -28,6 +28,12 @@ const paths = require('./paths');
 const getClientEnvironment = require('./env');
 const libs = require('./libs');
 
+// shuold we enable split chunking or not
```
Comment (joshsylvester) [inline packages/react-scripts/config/webpack.config.prod.js]: minor: "should"

Comment (elvinfucom) [reply packages/react-scripts/config/webpack.config.prod.js]: Thanks! Updated.

**Comment by elvinfucom** [comment]
2 approves, merge for now, please let me know if need some update.

**Comment by shinxi** [comment]
Thanks @elvinfucom. Two comments.
1. For proxySetup, actually this has been included in facebook/create-react-app, @joshsylvester any plan upgrade our create-react-app to include latest feature of facebook/create-react-app? We are far behind them. :)
2. @elvinfucom For `ENABLE_SPLIT_CHUNKS`, `GENERATE_RUNTIME_CHUNK`, curious why we want to disable code splitting, I saw one open issue from https://github.com/facebook/create-react-app/issues/5306, some people asked for some reasons, but what's our reason/issue to disable code splitting?

**Comment by joshsylvester** [comment]
@shinxi yes, we are a bit behind them - last time i approached upgrading to their setup (which now uses a single config file for webpack, most notably) it broke all builds, and all usage with our customizations. I spent almost a week trying to get it to build properly, but was unable to. upgrading to the latest of create-react-app is low priority, but we'd welcome the help if you have the bandwidth.

**Comment by shinxi** [comment]
@joshsylvester thanks the update, understand it's not easy. No bandwidth for me so far in 19.2, :( but we can see if have in 19.3.

**Pull request merged**